airflow-core/src/airflow/dag_processing/dagbag.py

```diff
@@ -647,9 +647,16 @@ def collect_dags(
                 found_dags = self.process_file(filepath, only_if_updated=only_if_updated, safe_mode=safe_mode)
 
                 file_parse_end_dttm = timezone.utcnow()
+                # Normalize paths to use forward slashes for consistent cross-platform behavior.
+                # On Windows, settings.DAGS_FOLDER and filepath may have mixed separators
+                # (e.g., DAGS_FOLDER='C:/foo/dags' vs filepath='C:\foo\dags\my_dag.py'),
+                # causing the replace to fail.
+                normalized_filepath = Path(filepath).as_posix()
+                normalized_dags_folder = Path(settings.DAGS_FOLDER).as_posix()
```

Comment:
I think just using `Path(filepath).relative_to(Path(settins.DAGS_FOLDER)).asPosix()` is much better.

Reply:
Thank you for the suggestion - fixed. Followup question: the existing code produces paths starting with `/`, whereas `relative_to` doesn't have that. This could be a breaking change for some users. Should I create a newsfragment in this case?

Reply:
Technically speaking, starting with "/" idepends on whether your DAGS_FOLDER was defined with "/" or not at the end.

I think this was ambiguous and **wrong** - precisely because it was not defined precisely, and because "/" at the beginning of a path universally means "start with root" - not "relative to DAGS FOLDER" which I guess was the original intention.

So yes, I think both clarifying what FileStat produces and adding newsfragment about it makes sense.

Reply:
newsfragment added